languages/common-lisp/exercises/concept/sameness/.docs/hints.md

```diff
@@ -1,23 +1,53 @@
+## General
+
+- Eli Bendersky has written [a good page][eli-lisp-equality] explaining the differences between the generic equality predicates in Common Lisp.
+- The [Hyperspec][hyperspec] is also a good reference and has pages defining each of the predicates: [`eq`][hyperspec-eq], [`eql`][hyperspec-eql], [`equal`][hyperspec-equal], and [`equalp`][hyperspec-equalp]
+- Remember that the four predicates range from most restrictive to most permissive in order: `eq`, `eql`, `equal`, `equalp`.
+
 ## 1. The maze of object equality
 
-This maze needs the most restrictive equality predicate available.
+- This maze needs the most restrictive equality predicate available.
+- [Eli Bendersky's page about Common Lisp equality][eli-lisp-equality] is a good explanation.
+- [The Hyperspec reference page for `eq`][hyperspec-eq] may be helpful.
```

Comment:
I've seen plenty of hints files that do this - just say: "you will need to use the `mcguffin [(link to reference page)]` for this one."  

that being said - I'm not entirely happy with it myself - it does seem too much like giving them the answer rather than "hinting".

Reply:
Hmm, okay. Should we open an issue for deciding what to do with this? I agree I've seen in in a number of places, but it does seem to give it away a bit.

Maybe there is some list of the documentation pages of the equality predicates to link to?

Reply:
I don't think there is a page in the Hyperspec for all the equality predicates. Eli's page I think it is a good reference.

Reply:
> This may be a bit too close to giving away the answer, but I'm not sure what crosses that line.

This is actually totally fine, as this is almost guaranteeing to unblock the student. And indeed, many other tracks do this. If there is a resource that has a nice description of a certain solution (instead of a man page), that _is_ usually preferable, but if there is no such page available, the man page link is perfect.